tests/ctst/steps/utils/kubernetes.ts

```diff
@@ -84,3 +123,174 @@ export async function createJobAndWaitForCompletion(world: Zenko, jobName: strin
         throw err;
     }
 }
+
+export async function waitForZenkoToStabilize(world: Zenko, needsReconciliation = false, namespace = 'default') {
+    // ZKOP pulls the overlay configuration from Pensieve every 5 seconds
+    // So the status might not be updated immediately after the overlay is applied.
+    // So, this function will first wait till we detect a reconciliation
+    // (deploymentInProgress = true), and then wait for the status to be available
+    const timeout = 15 * 60 * 1000;
```

Comment:
why 15 minutes ? can we make it configurable ? The default TO in cucumber is 100 seconds, should we make it 100 seconds as well by default ?

Reply:
I think we can access the timeout from the test by default, by passing the test object in the function, so that we wait just what's needed

Reply:
So, to answer the question: the functions will have the timeout as a parameter, but the default remains 15min; the reason is that the service restart can take up to 3~4min as per my observations, and sometimes longer, if some tests are running at high frequency, blocking kube from restarting the pods quickly (as they are still handling requests).
So, the steps reyling on these functions will need specific timeouts.